File: CLIESC_TICKETERA_SOAP_JAVA_GR01/src/renderer/src/mappers/Location.mapper.ts
import { Location } from '@renderer/entities/Location'
import { LocationResponse, Mapper } from '@renderer/types'

export class LocationMapper implements Mapper<LocationResponse, Location> {
  toEntity(dto: LocationResponse) {
    return new Location(dto.codigoLocalidad, dto.disponibilidad, dto.precio)
  }
}

File: CLICON_TICKETERA_SOAP_JAVA_GR01/src/main/java/ec/edu/monster/model/mapper/LocationMapper.java
package ec.edu.monster.model.mapper;

import ec.edu.monster.model.dto.LocationResponse;
import ec.edu.monster.model.entity.Location;

public class LocationMapper {
    public static Location toEntity(LocationResponse dto) {
        if (dto == null)
            return null;

        return new Location(
                dto.codigoLocalidad(),
                dto.disponibilidad(),
                dto.precio());
    }
}